src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -97,25 +113,31 @@ CHIP_ERROR ReadDerUnsignedIntegerIntoRaw(Reader & reader, uint8_t * raw_integer_
     // We now have the rest of the tag that is a "minimal length" unsigned integer.
     // Blit it at the correct offset, since the order we use is MSB first for
     // both ASN.1 and EC curve raw points.
-    size_t offset = raw_integer_length - integer_len;
-    return reader.ReadBytes(&raw_integer_out[offset], integer_len).StatusCode();
+    size_t offset = raw_integer_out.size() - integer_len;
+    return reader.ReadBytes(raw_integer_out.data() + offset, integer_len).StatusCode();
 }
 
-size_t EmitDerIntegerFromRaw(const uint8_t * raw_integer, size_t raw_integer_length_bytes, uint8_t * out_der_integer,
-                             size_t out_der_integer_size)
+/**
+ * @brief Utility to emit a DER-encoded INTEGER given a raw unsigned large integer
+ *        in big-endian order.
+ * @param[in] raw_integer Bytes of a large unsigned integer in big-endian, possibly including leading zeroes
+ * @param[out] out_der_integer Buffer to receive the DER-encoded integer
+ * @return The non-zero length used by DER encoding in `out_der_ingeger`, or 0 on error, such as value too large to fit output.
+ */
+size_t EmitDerIntegerFromRaw(const ByteSpan & raw_integer, MutableByteSpan out_der_integer)
 {
-    if ((raw_integer == nullptr) || (raw_integer_length_bytes == 0))
+    if ((raw_integer.empty()) || (raw_integer.size() == 0))
     {
         return 0;
     }
 
-    Reader reader(raw_integer, static_cast<uint16_t>(raw_integer_length_bytes));
-    BufferWriter writer(out_der_integer, static_cast<uint16_t>(out_der_integer_size));
+    Reader reader(raw_integer);
+    BufferWriter writer(out_der_integer);
 
     bool needs_leading_zero_byte = false;
 
     uint8_t cur_byte = 0;
-    while ((reader.Remaining() > 0) && (reader.Read8(&cur_byte).StatusCode() == CHIP_NO_ERROR) && (cur_byte == 0))
+    while ((reader.Read8(&cur_byte).StatusCode() == CHIP_NO_ERROR) && (cur_byte == 0))
```

Comment:
This is now assuming that if we get here and `Remaining()` is 0 (after going through the loop for a bit) then the failing `Read8` call will not modify `cur_byte`.  I don't think `Read8` promises that, so the `Remaining` check should stay here, I think.  Or we need to document on the reader that `Read8` does not change its outparam on failure?

Reply:
This is why I originally had the .Remaining() call. It avoided having to consider those side effects and relying on error-handling of reader.

Reply:
Yes, and we should keep the `Remaining` here and document that it's to avoid having to think about that.  I thought I said that in my comments on #8236....  It's the other callsite where removing `Remaining` is fine.